hpcc-nagios-tools/HPCCNagiosToolSet.cpp

```diff
@@ -431,17 +442,39 @@ bool CHPCCNagiosToolSet::generateHostGroupsConfigurationFile(const char* pOutput
                     strcpy(pLastHostName,pHostName);
                 }
             }
+            arrIPs.appendUniq(pHostName);
         }
 
-        pch = strtok(NULL, ",\n");
+        pch = strtok(nullptr, ",\n");
 
         nCount++;
     }
 
     if (strHostConfig.length() > 0)
     {
-        appendHeader(strHostConfig);        
-        strHostConfig.append(P_NAGIOS_HOSTS_GROUP_END_BRACKET);
+        if (m_bGenerateCatchAllHostGroup)
+        {
+            strHostConfig.clear();
+            appendHeader(strHostConfig);
+
+            int len = arrIPs.length();
+            strHostConfig.append(P_NAGIOS_HOSTS_GROUP_CONFIG_1).append(CHPCCNagiosToolSet::m_CatchAllHostGroupName).append(P_NAGIOS_HOSTS_GROUP_ALIAS).append(CHPCCNagiosToolSet::m_CatchAllHostGroupAlias).append(P_NAGIOS_HOSTS_GROUP_MEMBERS);
+
+            while (len > 0)
+            {
+                strHostConfig.append(arrIPs[--len]);
+
+                if (len == 0)
+                    strHostConfig.append(P_NAGIOS_HOSTS_GROUP_END_BRACKET);
+                else
+                    strHostConfig.append(", ");
+            }
+        }
+        else
+        {
+            appendHeader(strHostConfig);
```

Comment:
Question: if (m_bGenerateCatchAllHostGroup), you clean  the strHostConfig before appendHeader(), but not here. Is it intentional?